nebula/ui/components/VPNComposerView.qml

```diff
@@ -0,0 +1,178 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import QtQuick 2.15
+import QtQuick.Layouts 1.15
+import Qt.labs.qmlmodels
+
+import Mozilla.VPN 1.0
+import Mozilla.VPN.qmlcomponents 1.0
+import components 0.1
+
+ColumnLayout {
+    id: composer
+    property var addon
+    required property int view
+
+    enum View {
+        Guide = 0,
+        Message = 1
+    }
+
+    spacing: 0
+
+    Repeater {
+        id: repeater
+        property var indentTagRegex: /(<\/?li>)/g
+        property bool isFirstButton: true
+
+        model: addon.composer.blocks
+        delegate: Loader {
+            id: loader
+
+            property var composerBlock: modelData
+
+            Layout.fillWidth: true
+            Layout.preferredHeight: item.implicitHeight
+
+            sourceComponent: getSourceComponent()
+
+            Component.onCompleted: {
+                //Avoids binding loops
+                var margin = getTopMargin()
+                if (modelData instanceof VPNComposerBlockButton) repeater.isFirstButton = false
+                Layout.topMargin = margin
+            }
+
+
+            function getSourceComponent() {
+                if (modelData instanceof VPNComposerBlockTitle) return titleBlock
+                if (modelData instanceof VPNComposerBlockText) return textBlock
+                if (modelData instanceof VPNComposerBlockOrderedList ||
+                    modelData instanceof VPNComposerBlockUnorderedList) return listBlock
+                if (modelData instanceof VPNComposerBlockButton)
+                    return composerBlock.style === VPNComposerBlockButton.Link ? linkButtonBlock : buttonBlock
```

Comment:
what about if here we return an error somehow? At least a `console.error()`

Reply:
I'd lean towards not showing the message if one of the blocks doesn't load... We'd have to come up with some UI/UX error state for that case since them message would still be visible in the inbox because we wouldn't perform this check until the message has been clicked. Putting a console log for now